src/cbass_viz.cpp

```diff
@@ -172,6 +172,10 @@ Rcpp::List CBASS_VIZcpp(const Eigen::VectorXd& x,
     //
     //
     while(rep_iter){
+      // Safety check - only so many iterations of the inner loop before we move on
+      if(try_iter > ti){
+        rep_iter = false;
```

Comment:
Using a `break` statement here may be cleaner than checking for `rep_iter` in every branch below.